mllib/src/main/scala/org/apache/spark/ml/tree/treeParams.scala

```diff
@@ -252,6 +253,35 @@ private[ml] object TreeRegressorParams {
   final val supportedImpurities: Array[String] = Array("variance").map(_.toLowerCase)
 }
 
+private[ml] trait DecisionTreeRegressorParams extends DecisionTreeParams
+  with TreeRegressorParams with HasVarianceCol {
+
+  /** @group setParam */
+  def setVarianceCol(value: String): this.type = set(varianceCol, value)
```

Comment:
This needs to be in the classes, not a trait, for Java compatibility.  I'd also note here that this is the biased sample variance (normalizing by number of samples n, rather than n-1): [https://en.wikipedia.org/wiki/Variance#Sample_variance]